pkg/virt-controller/network/ipamclaims_test.go

```diff
@@ -0,0 +1,198 @@
+	nadNetworkName         = "nad_network_name"
+)
+
+var _ = Describe("CreateIPAMClaims", func() {
+	var virtClient *kubecli.MockKubevirtClient
+	var networkClient *fakenetworkclient.Clientset
+	var vmi *virtv1.VirtualMachineInstance
+
+	BeforeEach(func() {
+		ctrl := gomock.NewController(GinkgoT())
+		virtClient = kubecli.NewMockKubevirtClient(ctrl)
+
+		networkClient = fakenetworkclient.NewSimpleClientset()
+		virtClient.EXPECT().NetworkClient().Return(networkClient).AnyTimes()
+		ipamClaimsClient := fakeipamclaimclient.NewSimpleClientset()
+		virtClient.EXPECT().IPAMClaimsClient().Return(ipamClaimsClient).AnyTimes()
+	})
+
+	BeforeEach(func() {
+		vmi = api.NewMinimalVMI(vmiName)
+		vmi.Namespace = namespace
+		vmi.Spec.Domain.Devices.Interfaces = []virtv1.Interface{
+			{Name: redNetworkLogicalName},
+			{Name: blueNetworkLogicalName},
+			{Name: "absent", State: virtv1.InterfaceStateAbsent},
+		}
+		vmi.UID = vmiUID
+		vmi.Spec.Networks = []virtv1.Network{
+			logicalSecondaryNetwork(redNetworkLogicalName, redNetworkNadName),
+			logicalSecondaryNetwork(blueNetworkLogicalName, blueNetworkNadName),
+			logicalSecondaryNetwork("absent", "absent-net"),
+		}
+	})
+
+	Context("With allowPersistentIPs enabled in the NADs", func() {
+		BeforeEach(func() {
+			persistentIPs := map[string]bool{redNetworkNadName: true, blueNetworkNadName: true}
+			Expect(createNADs(networkClient, namespace, vmi.Spec.Networks, persistentIPs)).ToNot(HaveOccurred())
+		})
+
+		It("should create the expected IPAMClaims", func() {
+			err := CreateIPAMClaims(virtClient, vmi, nil)
+			Expect(err).ToNot(HaveOccurred())
+
+			ipamClaimsList, err := virtClient.IPAMClaimsClient().K8sV1alpha1().IPAMClaims(namespace).List(
+				context.Background(),
+				v1.ListOptions{},
+			)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(ipamClaimsList.Items).To(HaveLen(2))
+			Expect(ipamClaimsList.Items[0].OwnerReferences).To(HaveLen(1))
+
+			assertIPAMClaim(ipamClaimsList.Items[0], blueNetworkLogicalName, "pod16477688c0e")
+			assertIPAMClaim(ipamClaimsList.Items[1], redNetworkLogicalName, "podb1f51a511f1")
+		})
+	})
+})
+
+var _ = Describe("GetNetworkToIPAMClaimParams", func() {
+	var virtClient *kubecli.MockKubevirtClient
+	var networkClient *fakenetworkclient.Clientset
+	var networks []virtv1.Network
+
+	BeforeEach(func() {
+		ctrl := gomock.NewController(GinkgoT())
+		virtClient = kubecli.NewMockKubevirtClient(ctrl)
+
+		networkClient = fakenetworkclient.NewSimpleClientset()
+		virtClient.EXPECT().NetworkClient().Return(networkClient).AnyTimes()
+	})
+
+	Context("With NADs", func() {
+		BeforeEach(func() {
+			networks = []virtv1.Network{
+				logicalSecondaryNetwork(redNetworkLogicalName, redNetworkNadName),
+				logicalSecondaryNetwork(blueNetworkLogicalName, blueNetworkNadName),
+			}
+
+			persistentIPs := map[string]bool{redNetworkNadName: true}
+			Expect(createNADs(networkClient, namespace, networks, persistentIPs)).ToNot(HaveOccurred())
```

Comment:
expect to succeed, instead of not having errors please. Positive logic for the win.